File: DesignPatternsDotNet/CommandPattern/Commands/TurnMobileOff.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CommandPattern.Commands
{
    public class TurnMobileOff : ICommand{
	private IReceiver myReceiver  = null;
	public TurnMobileOff(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void Execute() {
		myReceiver.Off();	
	}
	public void Undo() {
		myReceiver.On();
	}
}
}

File: DesignPatternsJava/src/com/command/TurnMobileOff.java
package com.command;

public class TurnMobileOff implements ICommand{
	private IReceiver myReceiver  = null;
	public TurnMobileOff(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void execute() {
		myReceiver.off();	
	}
	public void undo() {
		myReceiver.on();
	}
}
